types/react-native/index.d.ts

```diff
@@ -4955,7 +4954,9 @@ export namespace StyleSheet {
      * their respective objects, merged as one and then returned. This also explains
      * the alternative use.
      */
-    export function flatten(style?: Style | Style[]): Style
+    export function flatten(style?: StyleProp<ViewStyle>): ViewStyle
+    export function flatten(style?: StyleProp<TextStyle>): TextStyle
+    export function flatten(style?: StyleProp<ImageStyle>): ImageStyle
```

Comment:
It doesn't make sense for `flatten` to operate on union type since you shouldn't be able to mix `ViewStyle`, `TextStyle` and `ImageStyle` for the same component.